ui/src/layouts/chats/presentation/messages/mod.rs

```diff
@@ -575,10 +579,39 @@ fn render_message<'a>(cx: Scope<'a, MessageProps<'a>>) -> Element<'a> {
         .as_ref()
         .unwrap_or(&msg_lines)
         .clone();
+    let preview_file_in_the_message: &UseState<(bool, Option<File>)> =
+        use_state(cx, || (false, None));
 
     cx.render(rsx!(
         div {
             class: "msg-wrapper",
+            preview_file_in_the_message.0.then(|| {
+                let file = preview_file_in_the_message.1.clone().map(|f| f.clone()).unwrap();
+                let file2 = file.clone();
+                rsx!(open_file_preview_modal {
+                    on_dismiss: |_| {
+                        preview_file_in_the_message.set((false, None));
+                    },
+                    on_download: move |temp_path: Option<PathBuf>| {
+                        let conv_id = message.inner.conversation_id();
+                        if let Some(path) = temp_path {
+                            if !path.exists() {
+                                log::info!("downloading file in temp directory: {:?}", path.clone());
+                                ch.send(MessagesCommand::DownloadAttachment {
+                                    conv_id,
+                                    msg_id: message.inner.id(),
+                                    file: file2.clone(),
+                                    file_path_to_download: path,
+                                })
+                            }
+                        } else {
+                            download_file(file2.clone(), message.inner.conversation_id(), message.inner.id(), pending_downloads, ch.clone());
```

Comment:
Not that its required but maybe we can have `download_file` use a reference for `File` and `Coroutine` (assuming its send function isnt mutable). Would make it cleaner than having it cloned and usually better to borrow if possible (although there is no breakage for it to be cloned either). See `download_file` for suggestion